### PR TITLE
Allow using localized absolute date times within phrases with place holders and localize issue due date events

### PR DESCRIPTION
**Pull request by yardenshoham**
This refactors the `shared/datetime/short|long|full` templates into a template helper function, which allows us to render absolute date times within translatable phrases.

- Follows #23988
- The first attempt was in #24055
- This should help #22664 

Changes:
1. Added the `DateTime` template helper that replaces the `shared/datetime/short|long|full` templates
2. Used find-and-replace with varying regexes to replace the templates from step 1 (for example, `\{\{template "shared/datetime/(\S+) \(dict "Datetime" ([^"]+) "Fallback" ([^\)]+\)?) ?\)?\}\}` -> `{{DateTime "$1 $2 $3}}`)
3. Used the new `DateTime` helper in the issue due date timestamp rendering

# Before
image

# After
image

### CODE REVIEW

File: modules/templates/helper.go [hunk 1]
```diff
@@ -135,6 +135,7 @@ func NewFuncMap() []template.FuncMap {
 		"CountFmt":      base.FormatNumberSI,
 		"TimeSince":     timeutil.TimeSince,
 		"TimeSinceUnix": timeutil.TimeSinceUnix,
+		"DateTime":      timeutil.DateTime,
```
Comment (wxiaoguang) [inline modules/templates/helper.go:142]: What do you think about #24266 ? Then we can have a `TimeUtils.DateDate`, and move all date/time related helper functions into `TimeUtils`, to simplify the helper.go.

File: modules/timeutil/datetime.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// DateTime renders an absolute time HTML given a time as a string
+func DateTime(format, datetime, fallback string) template.HTML {
+	switch format {
+	case "short":
+		return template.HTML(fmt.Sprintf(`<relative-time format="datetime" year="numeric" month="short" day="numeric" weekday="" datetime="%s">%s</relative-time>`, datetime, fallback))
+	case "long":
+		return template.HTML(fmt.Sprintf(`<relative-time format="datetime" year="numeric" month="long" day="numeric" weekday="" datetime="%s">%s</relative-time>`, datetime, fallback))
+	case "full":
+		return template.HTML(fmt.Sprintf(`<relative-time format="datetime" weekday="" year="numeric" month="short" day="numeric" hour="numeric" minute="numeric" second="numeric" datetime="%s">%s</relative-time>`, datetime, fallback))
+	}
```
Comment (wxiaoguang) [inline modules/timeutil/datetime.go]: Since `datetime` and `fallback` are all passed in as string, they need to be escaped before being used in HTML.

Comment (yardenshoham) [reply modules/timeutil/datetime.go]: What's the recommended way to do that?

Comment (wxiaoguang) [reply modules/timeutil/datetime.go]:
```
datetimeEscaped := html.EscapeStirng(datetime)
fallbackEscaped := html.EscapeStirng(fallback)
Sprintf("... %s ... %s ...", datetimeEscaped, fallbackEscaped)
```

File: modules/templates/helper.go (hunk 1 above)
Comment (yardenshoham) [reply modules/templates/helper.go:142]: Whichever PR lands first will adopt the other

Comment (yardenshoham) [reply modules/templates/helper.go:142]: Looks like #24266 is already in the merge queue. I think that moving all date/time related should happen in a PR of its own

Comment (silverwind) [reply modules/templates/helper.go:142]: https://github.com/go-gitea/gitea/pull/24266 is in.

Comment (yardenshoham) [reply modules/templates/helper.go:142]: Yes but I don't want to bloat this PR

**Review by silverwind — APPROVED** [review]
Seems to work well.

File: modules/templates/helper.go (hunk 1 above)
Comment (silverwind) [reply modules/templates/helper.go:142]: ok

File: templates/repo/issue/view_content/comments.tmpl [hunk 3]
```diff
@@ -314,7 +316,7 @@
 				{{template "shared/user/avatarlink" dict "Context" $.Context "user" .Poster}}
 				<span class="text grey muted-links">
 					{{template "shared/user/authorlink" .Poster}}
-					{{$.locale.Tr "repo.issues.due_date_remove" .Content $createdStr | Safe}}
+					{{$.locale.Tr "repo.issues.due_date_remove" (DateTime "long" .Content .Content) $createdStr | Safe}}
```
Comment (wxiaoguang) [inline templates/repo/issue/view_content/comments.tmpl:319]: It is `short` format in database?

**Review by wxiaoguang — APPROVED** [review]

File: templates/repo/issue/view_content/comments.tmpl (hunk 3 above)
Comment (yardenshoham) [reply templates/repo/issue/view_content/comments.tmpl:319]: long means full month name (e.g. April), short means short month name (e.g. Apr)

**Comment by silverwind** [comment]
.

**Pull request merged**